File: pm_core_1/src/infrastructure/store/model/priority_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

type PriorityModel struct {
	BaseModel
	TenantID    int64   `gorm:"column:tenant_id;not null" json:"tenantId"`
	Name        string  `gorm:"column:name;type:varchar(100);not null" json:"name"`
	Description *string `gorm:"column:description" json:"description,omitempty"`
	IconUrl     *string `gorm:"column:icon_url" json:"iconUrl,omitempty"`
	Color       *string `gorm:"column:color;type:varchar(20)" json:"color,omitempty"`
	Sequence    int     `gorm:"column:sequence;not null" json:"sequence"`
	IsSystem    bool    `gorm:"column:is_system;not null" json:"isSystem"`
}

func (PriorityModel) TableName() string {
	return "priorities"
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/model/PriorityModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "priorities")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class PriorityModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "icon_url")
    private String iconUrl;

    @Column(name = "color")
    private String color;

    @Column(name = "sequence")
    private Integer sequence;

    @Column(name = "is_system")
    private Boolean isSystem;
}
